tap_postgres/sync_strategies/incremental.py

```diff
@@ -119,9 +119,40 @@ def sync_table(conn_info, stream, state, desired_columns, md_map):
                         singer.write_message(singer.StateMessage(value=copy.deepcopy(state)))
 
                     counter.increment()
+                
+                if rows_saved == 0:
+                    latest_sql = _get_select_latest_sql({"escaped_columns": escaped_columns,
+                                              "replication_key": replication_key,
+                                              "schema_name": schema_name,
+                                              "table_name": stream['table_name'],
+                                            })
+                    LOGGER.info('select latest row statement: %s', latest_sql)
+                    cur.execute(select_sql)
+                    for rec in cur:
```

Comment:
```suggestion
                    rec = cur.fetchone()
```

**Suggestion:** No need a for-loop here as the query always returns a single record

Reply:
Just resolved this in my latest commit sir 😄 .